framework/aster-frame/src/arch/x86/trap.rs

```diff
@@ -47,6 +48,40 @@ extern "sysv64" fn trap_handler(f: &mut TrapFrame) {
                 handle_virtual_exception(f, &ve_info);
             }
             &PAGE_FAULT => {
+                let page_fault_addr = unsafe { x86::controlregs::cr2() };
+
+                // Handle page fault from user space.
+                if (0..MAX_USERSPACE_VADDR).contains(&page_fault_addr) {
+                    let current_task = current_task().unwrap();
+                    let Some(user_space) = current_task.user_space() else {
+                        return;
```

Comment:
Seems a bad state here. Panic instead of returning from the interrupt context, which is bad, solving nothing, and make error prompts vague.